sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
+            
+            _updateCompetencySkillsFn = function(updatedDemonstrationSkills) {
+                // render details tables
+                updatedDemonstrationLength = updatedDemonstrationSkills.length;
+                updatedDemonstrationIndex = 0;
+                
+                for (; updatedDemonstrationIndex < updatedDemonstrationLength; updatedDemonstrationIndex++) {
+
+                    demonstratedSkill = updatedDemonstrationSkills[updatedDemonstrationIndex];
+                    skill = competencySkills.get(demonstratedSkill.SkillID);
+                    
+                    if (skill) {
+                        demonstrationsByStudent = skill.demonstrationsByStudent || (skill.demonstrationsByStudent = {});
+
+                        if (demonstratedSkill.StudentID in demonstrationsByStudent) {
+                            demonstrationsByStudent[demonstratedSkill.StudentID].push(demonstratedSkill);
+                        } else {
+                            demonstrationsByStudent[demonstratedSkill.StudentID] = [demonstratedSkill];
+                        }
+                    }
+
+                } 
+
+                skillStudentsTpl.overwrite(skillStudentsRow.down('tbody'), {
+                    skills: competencySkills.items,
+                    studentIds: Ext.Object.getKeys(update.competency.get('studentCompletions'))
+                });
+                
+                me.syncRowHeights(
+                    skillHeadersRow.select('tr'),
+                    skillStudentsRow.select('tr')
+                );
+            };
+
+            for (; updateIndex < updatesLength; updateIndex++) {
+                update = updatesCompetencies[updateIndex];
+
+                competencySkills = update.competency.get('skills');
+                
+                if (update.competency.get('expanded')) {
+                    competencySkills.each(_compentencyDemonstrationResetFn);
+                           
+                    skillsRows = dashboardView.el.select(
+                        Ext.String.format('.cbl-grid-skills-row[data-competency="{0}"]', update.competency.getId()),
+                        true // true to get back unique Ext.Element instances
+                    );
+                    
+                    console.log(skillsRows);
+                    
+                    skillHeadersRow = skillsRows.item(0);
+                    skillStudentsRow = skillsRows.item(1);
+    
+                    Ext.fly(update.textDom).update(update.percent+'%');
+                    Ext.fly(update.barDom).setWidth(update.percent+'%');
+                    Ext.fly(update.averageDom).update(Ext.util.Format.number(update.average, '0.##'));
+                    Ext.fly(update.cellDom).toggleCls('is-average-low', update.isAverageLow);
+                    
+                    update.competency.getDemonstrationsForStudents(Ext.pluck(SiteEnvironment.cblStudents, 'ID'), _updateCompetencySkillsFn, me);
```

Comment:
Is this ``pluck`` necessary. Could we have been building this value as we went about our business above?